src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public void fill(List<? super Machine> machines, Object value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, (Machine)value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> list) {
+        for (Machine machine : list) {
+            machine.doWork();
+        }
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<Bulldozer> machines = bulldozerProducer.get();
+            return new ArrayList<>(machines);
```

Comment:
Redundant variables, can be simplified (as well as in other cases in this method)